server/src/main/java/org/apache/druid/server/http/MetadataResource.java

```diff
@@ -471,4 +471,23 @@ public Response getDataSourceInformation(
     );
     return Response.status(Response.Status.OK).entity(authorizedDataSourceInformation).build();
   }
+
+  /**
+   * @return all bootstrap segments determined by the coordinator.
+   */
+  @POST
+  @Path("/bootstrapSegments")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(DatasourceResourceFilter.class)
+  public Response getBootstrapSegments()
+  {
+    final Set<DataSegment> broadcastSegments = coordinator.getBroadcastSegments();
+    if (broadcastSegments == null) {
+      return Response.status(Response.Status.SERVICE_UNAVAILABLE)
+                     .entity("No bootstrap segments were found. The coordinator may still be"
```

Comment:
Nit: I think the error message would be more accurate if it said that the state of bootstrap segments was unknown, rather then "none were found" (which sounds like there might be an empty set, vs. the coordinator not providing the info)